sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureFileSystemProvider.java

```diff
@@ -269,7 +284,18 @@ public Path getPath(URI uri) {
     @Override
     public SeekableByteChannel newByteChannel(Path path, Set<? extends OpenOption> set,
             FileAttribute<?>... fileAttributes) throws IOException {
-        throw LoggingUtility.logError(logger, new UnsupportedOperationException());
+        if (Objects.isNull(set)) {
+            set = Collections.emptySet();
+        }
+
+        if (set.contains(StandardOpenOption.WRITE)) {
```

Comment:
Should we consider the `CREATE` or `CREATE_NEW` options as `WRITE`? Or are we expecting that either of these would be used in tandem with `WRITE`.

Reply:
According to the javadoc:  The READ and WRITE options determine if the file should be opened for reading and/or writing. If neither option (or the APPEND option) is present then the file is opened for reading. 

I take that to mean you have to specify write to open in write mode. But I'm open to discuss that.